src/persist-client/src/internal/metrics.rs

```diff
@@ -1654,6 +1670,185 @@ impl SinkMetrics {
+            correction_sum_insertions: registry.register(metric!(
+                name: "mz_persist_sink_correction_sum_insertions",
+                help: "The cumulative insertions observed on the correction buffer across workers and persist sinks.",
+            )),
+            correction_sum_deletions: registry.register(metric!(
+                name: "mz_persist_sink_correction_sum_deletions",
+                help: "The cumulative deletions observed on the correction buffer across workers and persist sinks.",
+            )),
+            correction_sum_capacity_increases: registry.register(metric!(
+                name: "mz_persist_sink_correction_sum_capacity_increases",
+                help: "The cumulative capacity increases observed on the correction buffer across workers and persist sinks.",
+            )),
+            correction_sum_capacity_decreases: registry.register(metric!(
+                name: "mz_persist_sink_correction_sum_capacity_decreases",
+                help: "The cumulative capacity decreases observed on the correction buffer across workers and persist sinks.",
+            )),
+            correction_peak_len: registry.register(metric!(
+                name: "mz_persist_sink_correction_peak_len",
+                help: "The peak length observed for the correction buffer across workers and persist sinks.",
+            )),
+            correction_peak_capacity: registry.register(metric!(
+                name: "mz_persist_sink_correction_peak_capacity",
+                help: "The peak capacity observed for the correction buffer across workers and persist sinks.",
+            )),
+            correction_max_len_per_worker: registry.register(metric!(
+                name: "mz_persist_sink_correction_max_len_per_worker",
+                help: "The maximum length observed for the correction buffer of any single persist sink per worker.",
+                var_labels: ["worker_id"],
+            )),
+            correction_max_capacity_per_worker: registry.register(metric!(
+                name: "mz_persist_sink_correction_max_capacity_per_worker",
+                help: "The maximum capacity observed for the correction buffer of any single persist sink per worker.",
+                var_labels: ["worker_id"],
+            )),
+        }
+    }
+
+    /// Obtains a `SinkWorkerMetrics` instance, which allows for metric reporting
+    /// from a specific `persist_sink` instance for a given worker. The reports will
+    /// update metrics shared across workers, but provide per-worker contributions
+    /// to them.
+    pub fn for_worker(&self, worker_id: usize) -> SinkWorkerMetrics {
+        let worker = worker_id.to_string();
+        let correction_max_len_per_worker = self
+            .correction_max_len_per_worker
+            .with_label_values(&[&worker]);
+        let correction_max_capacity_per_worker = self
+            .correction_max_capacity_per_worker
+            .with_label_values(&[&worker]);
```

Comment:
Why do we only specialize some of the metrics per worker? The cost of having all per worker seems to be relatively low, doesn't it?

Reply:
I have the opposite inclination, fwiw! Grafana/Prometheus have seemed to be getting slower and slower for me as time goes on, so even if some new incremental metric load doesn't cause them to fall over, it does seem to be adding up. As a result, I try to start with the minimum set of metrics that I think will answer the questions I need answered, and then increasing granularity only if/when I hit questions that can't be answered with existing metrics

Reply:
For now, I am keeping the metrics computed exactly the same. I think it should be easy to refine this in subsequent PRs if we feel that we need to add more detail to the per-worker measurements?